docs/guides/database.rst

```diff
@@ -169,14 +169,36 @@ Entity Icons
 ~~~~~~~~~~~~
 
 Every entity can be assigned an icon which is retrieved using the ``ElggEntity::getIconURL($params)`` method.
-This method accepts a ``$params`` argument that can be either a string specifying on of the configured icon sizes,
-or an array of parameters, that specify the size and provide additional context for the hook to determine the icon
-to serve.
+This method accepts a ``$params`` argument as an array that specifies the size, type, and provide 
+additional context for the hook to determine the icon to serve.
 
-Use ``elgg_get_config('icon_sizes')`` to get all possible values. The following sizes exist by default:
+Use ``elgg_get_config('icon_sizes')`` to get all possible icon sizes. The following sizes exist by default:
 ``'large'``, ``'medium'``, ``'small'``, ``'tiny'``, and ``'topbar'``. The method triggers the
 ``entity:icon:url`` :ref:`hook <guides/hooks-list#other>`.
 
+Entity icons can be saved from uploaded files, existing local files, or  existing ElggFile 
+objects. These methods save all sizes of icons defined in the system.
+
+.. code:: php
+	$object = new ElggObject();
+	$object->title = 'Example entity';
+	$object->description = 'An example object with an icon.';
+	
+	// from an uploaded file
+	$tmp_name = _elgg_services()->request()->files->get('my_uploaded_file')->getPathName();
```

Comment:
Elgg\Request is also private, so we'd either need to change that or go with elgg_get_uploaded_file_by_name().

In general, I'm more for graduating services to public API than introducing more global functions.